### ISSUE TITLE
Relax `engines.node` to ">=4"?

### ISSUE THREAD

**Issue by evocateur** [sholladay/mkdirtemp#1]
I was excited to use this delightfully clean module, and then sadness intervened:
```sh
# OS X Sierra, bash 4
$ nvm current
v4.8.0
$ yarn add mkdirtemp
yarn add v0.21.3
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error mkdirtemp@1.0.0: The engine "node" is incompatible with this module. Expected version ">=6".
error Found incompatible module
```

I inadvertently had `nvm` pointing to v4.8.0 because a project I help maintain ([lerna](https://github.com/lerna/lerna)) still supports v4, and I was trying to figure out why all the v4 CI was failing (naturally, it was locally unreproducible 😭 ). I was able to workaround yarn's failure with the `--ignore-engines` flag, but that's less than ideal for everyone else maintaining the project...

In any case, looking at the code (again, which I really appreciate in terms of simplicity and elegance), I don't see anything that necessarily restricts the previous node LTS (v4) from being allowed:

```sh
$ nvm use v6
Now using node v6.9.4 (npm v3.10.10)
$ yarn add mkdirtemp
# ...
$ nvm exec v4 node -e 'require("mkdirtemp")().then(console.log)'
Running node v4.8.0 (npm v2.15.11)
/var/folders/9s/95tsbq5j12s5yqgxqp8cqb540000gp/T/3heHc9
```

Thanks!

**Comment by sholladay** [comment]
Hey @evocateur, glad you like it. :)

I would be happy to relax to Node 4. Just need to double check that everything I'm using is supported (arrow functions, etc) and make the necessary tweaks if not. Mind helping with that? Looks like it works from your `node -e` example.

It would be nice to have [CI test multiple Node versions](https://discuss.circleci.com/t/testing-multiple-versions-of-node/542). Otherwise I will be prone to breaking support by accident, as I tend to live on the bleeding edge. Not required, though.

**Comment by evocateur** [comment]
Sure thing! I'll have a PR tomorrow.

**Comment by sholladay** [comment]
Sorry @evocateur, it appears Node 4.x [isn't going to work](https://circleci.com/gh/sholladay/mkdirtemp/3). The [`fs.mkdtemp()`](https://nodejs.org/api/fs.html#fs_fs_mkdtemp_prefix_options_callback) method was [added in Node 5.10.0](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V5.md#2016-03-31-version-5100-stable-evanlucas). I have no idea why your `nvm` example worked for you! 😬 

So it appears the [best I can do](https://github.com/sholladay/mkdirtemp/commit/251f59b4d3865410a4dd3136193d93672d605145) is `>=5.10`. I'm still happy to do this, but you should be aware that the 5.x line went [end-of-life in mid 2016](https://github.com/nodejs/LTS/blob/0fafe811a73e4261d99f4b24dfc56a781fb8bbb2/schedule.png).

If that is painful for you, there are three options.

1. [Petition Node core](https://github.com/nodejs/node/issues/new) to backport `fs.mkdtemp()`. I'm not sure how willing they are to backport features during LTS mode, but Node 4.x is still supported so this seems possible.
2. Create an `fs.mkdtemp()` polyfill. I don't think it's possible to create a safe one in JS, but could possibly be done with a [native module](https://nodejs.org/api/addons.html). If someone did that I would consider using it here.
3. Look at other libraries. I contribute to Sindre's [temp-write](https://github.com/sindresorhus/temp-write) and can attest to its awesomeness. And [tempfile](https://github.com/sindresorhus/tempfile) is good when you only need a path. Just keep in mind that anything that doesn't use the UNIX [`mkdtemp`](http://pubs.opengroup.org/onlinepubs/9699919799/functions/mkdtemp.html) under the hood is not as safe (hence why I built this module).

**Comment by evocateur** [comment]
I... could have _sworn_ I checked >.<. I really appreciate your patience with my absent-minded debugging brain!

I will check out the others, my apologies for the awkwardness.

**Comment by sholladay** [comment]
Good news @evocateur! It already [is backported](https://github.com/nodejs/node/blob/3b05153cdccdaaa74e21414541d70de7937f7965/doc/changelogs/CHANGELOG_V4.md#2017-02-21-version-480-argon-lts-mylesborins).

I initially missed the fact that when Circle failed, it happened on Node 4.5. I assumed that your `nvm` command would work the same on Circle and on your machine. But no! Computers!

Circle has [some versions preinstalled](https://circleci.com/docs/1.0/build-image-trusty/#nodejs). The version that includes the `fs.mkdtemp()` backport, 4.8, is not currently among them. So we need to install it manually, otherwise `nvm` uses an outdated version.

I will have a release for you later this weekend. Thanks for your patience. 😄

**Comment by evocateur** [comment]
Frabjous day!

**Comment by sholladay** [comment]
Released with `mkdirtemp@1.1.0`. Enjoy! 😄

**Comment by evocateur** [comment]
Thanks!
